src/client.js

```diff
@@ -1287,36 +1281,27 @@ function SugarApi(args) {
          * @param {Object} [$files] jQuery/Zepto DOM elements that carry the files to upload.
          * @param {Object} [callbacks] callback object.
          * @param {Object} [options] Request options hash.
-         *
-         * - htmlJsonFormat: Boolean flag indicating if `sugar-html-json` format must be used (`true` by default)
-         * - iframe: Boolean flag indicating if iframe transport is used (`true` by default)
          * See {@link Api#buildFileURL} function for other options.
-         *
          * @return {HttpRequest} The AJAX request.
          * @memberOf Api
          * @instance
          */
         file: function(method, data, $files, callbacks, options) {
-            var ajaxParams = {
-                files: $files,
-                processData: false
+            let ajaxParams = {
+                processData: false,
+                contentType: false,
             };
 
-            //delete method doesn't need to go through the iframe transport
-            if (method === 'delete') {
-                ajaxParams.iframe = false;
-            } else if (!options || options.iframe !== false) {
-                ajaxParams.iframe = true;
-
-                // pass OAuth token as GET-parameter during file upload.
-                // otherwise, in case if file is too large, the whole request body may be
-                // ignored by interpreter together with the token
-                options = options || {};
-                options.passOAuthToken = true;
+            let fd = new FormData();
+
+            if (data.field && $files && !_.isUndefined($files[0]) && $files[0].files && $files[0].files.length > 0) {
+                fd.append(data.field, $files[0].files[0]);
             }
```

Comment:
One small thing here: the length check against 0 could be removed (any number greater than 0 should give a `true` and if it is 0 it should be automatically `false`).

Also I wonder if the if statement above could be brought under a more readable format, something like:
`
if (data.field && $files) { // we check if there was anything sent at all
    let attachedFile = $files[0];
    if (!_.isUndefined(attachedFile) && attachedFile.files && attachedFile.files.length) { //we check if we really have files to work with
        fd.append(data.field, attachedFile.files[0]);
    }
}
`